Client/Frontend/Browser/TabManager.swift

```diff
@@ -173,6 +173,11 @@ class TabManager: NSObject {
             return
         }
 
+        // Make sure to wipe the private tabs if the user has the pref turned on
+        if shouldClearPrivateTabs(), !(tab?.isPrivate ?? false) {
+            removeAllPrivateTabsAndNotify(false)
```

Comment:
> the previous tab is nil in a lot of cases. 

Is this true? I see at the top of this method:

```swift
    func selectTab(_ tab: Tab?, previous: Tab? = nil) {
        assert(Thread.isMainThread)
        let previous = previous ?? selectedTab
```

We can get our desired behaviour by:

```swift
         if !(tab?.isPrivate ?? false) {
            willSwitchTabMode()
        }
```